Purchases/Networking/Operations/NetworkOperation.swift

```diff
@@ -0,0 +1,67 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  NetworkOperation.swift
+//
+//  Created by Joshua Liebowitz on 11/18/21.
+
+import Foundation
+
+class NetworkOperation: Operation {
+
+    let httpClient: HTTPClient
+    let authHeaders: [String: String]
+
+    init(configuration: NetworkConfiguration) {
+        self.httpClient = configuration.httpClient
+        self.authHeaders = configuration.authHeaders
+
+        super.init()
+    }
```

Comment:
What do you think about implementing with `main` as `fatalError("Subclasses must override this method")`?

Reply:
🤦‍♂️ can't believe I missed this one.

Reply:
✅